htlcswitch/iterator.go

```diff
@@ -85,64 +82,32 @@ func (r *sphinxHopIterator) EncodeNextHop(w io.Writer) error {
 func (r *sphinxHopIterator) ForwardingInstructions() (
 	hop.ForwardingInfo, error) {
 
-	var (
-		nextHop lnwire.ShortChannelID
-		amt     uint64
-		cltv    uint32
-	)
-
 	switch r.processedPacket.Payload.Type {
 	// If this is the legacy payload, then we'll extract the information
 	// directly from the pre-populated ForwardingInstructions field.
 	case sphinx.PayloadLegacy:
 		fwdInst := r.processedPacket.ForwardingInstructions
+		p := hop.NewLegacyPayload(fwdInst)
 
-		switch r.processedPacket.Action {
-		case sphinx.ExitNode:
-			nextHop = hop.Exit
-		case sphinx.MoreHops:
-			s := binary.BigEndian.Uint64(fwdInst.NextAddress[:])
-			nextHop = lnwire.NewShortChanIDFromInt(s)
-		}
```

Comment:
Isn't this mapping needed anymore? Is the `Action` field required at all then anymore?

Reply:
nope, the action field is no longer required, it can be removed from lightning-onion

Reply:
i think we need to talk again about embedding lightning-onion in the lnd repo